File: Binary_Search/PHP/mitogh/binary_search.php
<?php
function binary_search($arr, $target){

    if(!is_array($arr) || !count($arr) || !is_numeric($target)){
        return -1;
    }

    $top = count($arr) - 1; 
    $bottom = 0; 

    while($bottom <= $top){
        $middle = ($top + $bottom) / 2; 

        if($arr[$middle] === $target){
            return $middle; 
        }else{
            if($target < $arr[$middle]){
                $top = $middle - 1;
            }else{
                $bottom = $middle + 1; 
            }
        }
    }
    return -1;
}

File: Binary_Search/Python/jiang42/binary_search.py
def binary_search(arr, target):
  low, high = 0, len(arr)-1
  while low < high:
    mid = (low + high)/2
    if arr[mid] == target:
      return mid
    elif arr[mid] > target:
      high = mid - 1
    else:
      low = mid + 1
  return high
